File: 2363-merge-similar-items/2363-merge-similar-items.swift
class Solution {
	func mergeSimilarItems(_ items1: [[Int]], _ items2: [[Int]]) -> [[Int]] {
		let n = items1.count
		var items1 = items1

		for item2 in items2 {
			var found = false
			for i in 0 ..< n {
				let item1 = items1[i]
				if item1[0] == item2[0] {
					found = true
					items1[i] = [item1[0], item1[1] + item2[1]]
				}

			}

			if !found {
				items1.append(item2)
			}
		}

		items1.sort { a, b in
			a[0] < b[0]
		}

		return items1
	}
}

File: 2363-merge-similar-items/2363-merge-similar-items.ts
function mergeSimilarItems(items1: number[][], items2: number[][]): number[][] {
  items1.sort((a, b) => a[0] - b[0]);
  items2.sort((a, b) => a[0] - b[0]);

  const result: number[][] = [];
  let i = 0;
  let j = 0;

  while (i < items1.length || j < items2.length) {
    if (i < items1.length && j < items2.length) {
      const item1 = items1[i];
      const item2 = items2[j];

      if (item1[0] === item2[0]) {
        result.push([item1[0], item1[1] + item2[1]]);
        i++;
        j++;
        continue;
      }
      if (item1[0] < item2[0]) {
        result.push(item1);
        i++;
        continue;
      }
      if (item1[0] > item2[0]) {
        result.push(item2);
        j++;
        continue;
      }
    }
    if (i < items1.length) {
      result.push(items1[i]);
      i++;
    }
    if (j < items2.length) {
      result.push(items2[j]);
      j++;
    }
  }

  return result;
}
